bin/check_ldap.rb

```diff
@@ -0,0 +1,21 @@
+# This is a rails runner that will print the status of the database
+# Possible outcomes:
+#
+#   * `LDAP_READY`: connection successfull
+#   * `LDAP_FAIL`: cannot authenticate within ldap server
+#   * `LDAP_DOWN`: cannot connect to the ldap server
+
+def ldap_ready?
+  ldap_config = Velum::LDAP.ldap_config
+  ldap = Net::LDAP.new(ldap_config)
+
+  if ldap.bind
+    puts "LDAP_READY"
+  else
+    puts "LDAP_FAIL"
+  end
+rescue Errno::ECONNREFUSED
```

Comment:
I'm afraid this is not enough. You should be using `Net::LDAP::Error`, since this is the error class that wraps all kinds of errors from LDAP connections.

Reply:
I saw `Deprecation warning: Net::LDAP::ConnectionRefused will be deprecated. Use Errno::ECONNREFUSED instead.` The reported error was a connection refused but if I recall correctly using the general ldap error exception also works.

Reply:
Then I'd say to keep them both.

Reply:
The authentication is probably not working because if I add `Net::LDAP::Error` velum doesn't start. So a generic error is happening. I'll try to debug a bit more.